Extension name

For discussion:
Extension name is libsmbclient, but libsmbclient.so is not a good name, and create confusion with the library.

Even "autotool" don't like it.

    $ touch libsmbclient.c
    $ make
    *** Warning: Linking the shared library libsmbclient.la against the loadable module
    *** libsmbclient.so is not portable!
    
    *** Warning: Linking the shared library libsmbclient.la against the loadable module
    *** libsmbclient.so is not portable!
    libtool: link: rm -fr  .libs/libsmbclient.la .libs/libsmbclient.lai .libs/libsmbclient.so
    libtool: link: cc -shared  -fPIC -DPIC  .libs/libsmbclient.o .libs/smb_streams.o   -Wl,-rpath -Wl,/work/GIT/libsmbclient-php/.libs -Wl,-rpath -Wl,/work/GIT/libsmbclient-php/modules /work/GIT/libsmbclient-php/.libs/libsmbclient.so  -O2 -m64 -mtune=generic -Wl,-z -Wl,relro   -Wl,-soname -Wl,libsmbclient.so -o .libs/libsmbclient.so
    cc: error: /work/GIT/libsmbclient-php/.libs/libsmbclient.so: No such file or directory
    Makefile:188: recipe for target 'libsmbclient.la' failed
    make: *** [libsmbclient.la] Error 1


Ok, this can be workaround (make clean && make, and auto-provides filtering for downstream packaging)

Possible names:
* smb
* smbc
* smbclient (functions prefix)
* samba

--- a/.gitignore
+++ b/.gitignore
@@ -1,1 +1,54 @@
+# Local config
 phpunit.xml
+
+# Object files
+*.o
+*.lo
+
+# Libraries
+*.lib
+*.a
+*.la
+
+# Shared objects (inc. Windows DLLs)
+*.dll
+*.so
+*.so.*
+*.dylib
+
+# Executables
+*.exe
+*.out
+*.app
+
+# autotools
+.deps
+.libs
+config.cache
+config.guess
+config.h
+config.h.in
+config.log
+config.nice
+config.status
+config.sub
+configure
+configure.in
+conftest
+conftest.c
+Makefile
+Makefile.fragments
+Makefile.global
+Makefile.objects
+acinclude.m4
+aclocal.m4
+autom4te.cache
+build
+install-sh
+libtool
+ltmain.sh
+ltmain.sh.backup
+missing
+mkinstalldirs
+modules
+run-tests.php

--- a/.travis.yml
+++ b/.travis.yml
@@ -33,5 +33,5 @@ script:
   - ./configure
   - make
   - sudo make install
-  - echo 'extension="libsmbclient.so"' | sudo tee -a /etc/php5/cli/php.ini
+  - echo 'extension="smbclient.so"' | sudo tee -a /etc/php5/cli/php.ini
   - /usr/bin/php phpunit.phar --verbose

--- a/Makefile.in
+++ b/Makefile.in
@@ -1,9 +1,0 @@
-LTLIBRARY_NAME    = libsmbclient.la
-LTLIBRARY_SOURCES = libsmbclient.c
-LTLIBRARY_LIBADD  = $(LIBSMBCLIENT_LIBADD)
-LTLIBRARY_SHARED_NAME    = libsmbclient.la
-LTLIBRARY_SHARED_LIBADD  = $(LIBSMBCLIENT_SHARED_LIBADD)
-
-SUBDIRS = $(LIBSMBCLIENT_SUBDIRS)
-
-include $(top_srcdir)/build/dynlib.mk

--- a/README.md
+++ b/README.md
@@ -43,7 +43,7 @@ make install INSTALL_ROOT=/tmp/smbc
 - Activate libsmbclient-php in php.ini:
 
 ```sh
-extension="libsmbclient.so"
+extension="smbclient.so"
 ```
 
 Contributions and bug reports

--- a/config.m4
+++ b/config.m4
@@ -1,16 +1,12 @@
 dnl If your extension references something external, use with:
 
+PHP_ARG_ENABLE(libsmbclient, whether to enable smbclient support,
+[  --enable-smbclient      Enable libsmbclient support])
+
 PHP_ARG_WITH(libsmbclient, for libsmbclient support,
-[  --with-libsmbclient=DIR Include libsmbclient support.
+[  --with-libsmbclient=DIR Installation prefix of libsmbclient.
                           If DIR is not specified, use the system library.])
-
-dnl Otherwise use enable:
-
-dnl PHP_ARG_ENABLE(libsmbclient, whether to enable libsmbclient support,
-dnl Make sure that the comment is aligned:
-dnl [  --enable-libsmbclient           Enable libsmbclient support])
-
-if test "$PHP_LIBSMBCLIENT" != "no"; then
+if test "$PHP_SMBCLIENT" != "no"; then
 
   dnl If a path was given, expect to find the library in $PHP_LIBDIR
   dnl and the include file in include/
@@ -21,7 +17,7 @@ if test "$PHP_LIBSMBCLIENT" != "no"; then
 
     AC_DEFINE(HAVE_LIBSMBCLIENT, 1, [ ])
     PHP_ADD_INCLUDE($LIBSMBCLIENT_INCDIR)
-    PHP_ADD_LIBRARY_WITH_PATH(smbclient, $LIBSMBCLIENT_DIR, LIBSMBCLIENT_SHARED_LIBADD)
+    PHP_ADD_LIBRARY_WITH_PATH(smbclient, $LIBSMBCLIENT_DIR, SMBCLIENT_SHARED_LIBADD)
 
   dnl Otherwise find the header and shared library by the normal means:
   else
@@ -41,13 +37,13 @@ if test "$PHP_LIBSMBCLIENT" != "no"; then
     fi
 
     dnl Find the shared library:
-    PHP_CHECK_LIBRARY(smbclient, smbc_open,
+    PHP_CHECK_LIBRARY(smbclient, smbc_getOptionUserData,
     [
       AC_DEFINE(HAVE_LIBSMBCLIENT, 1, [ ])
       PHP_ADD_INCLUDE($LIBSMBCLIENT_INCDIR)
-      PHP_ADD_LIBRARY(smbclient, 1, LIBSMBCLIENT_SHARED_LIBADD)
+      PHP_ADD_LIBRARY(smbclient, 1, SMBCLIENT_SHARED_LIBADD)
     ],[
-      AC_MSG_ERROR([Could not find libsmbclient.so or symbol smbc_open. Check config.log for more information.])
+      AC_MSG_ERROR([Could not find libsmbclient.so or symbol smbc_getOptionUserData. Check version and config.log for more information.])
     ],[
       -lsmbclient
     ])
@@ -74,7 +70,7 @@ if test "$PHP_LIBSMBCLIENT" != "no"; then
     -lsmbclient
   ])
 
-  PHP_SUBST(LIBSMBCLIENT_SHARED_LIBADD)
-  PHP_NEW_EXTENSION(libsmbclient, libsmbclient.c smb_streams.c, $ext_shared)
+  PHP_SUBST(SMBCLIENT_SHARED_LIBADD)
+  PHP_NEW_EXTENSION(smbclient, smbclient.c smb_streams.c, $ext_shared)
 
 fi

--- a/php_smbclient.h
+++ b/php_smbclient.h
@@ -46,12 +46,12 @@
 #define PHP_SMBCLIENT_VERSION "0.8.0-dev"
 
 extern zend_module_entry smbclient_module_entry;
-#define phpext_libsmbclient_ptr &libsmbclient_module_entry
+#define phpext_smbclient_ptr &smbclient_module_entry
 
 typedef struct {
 } php_smbclient_globals;
 
-typedef struct _php_libsmbclient_state
+typedef struct _php_smbclient_state
 {
 	SMBCCTX *ctx;
 	char *wrkg;
@@ -62,7 +62,7 @@ typedef struct _php_libsmbclient_state
 	int passlen;
 	int err;
 }
-php_libsmbclient_state;
+php_smbclient_state;
 
 PHP_MINIT_FUNCTION(smbclient);
 PHP_MSHUTDOWN_FUNCTION(smbclient);
@@ -102,19 +102,19 @@ PHP_FUNCTION(smbclient_statvfs);
 PHP_FUNCTION(smbclient_fstatvfs);
 
 /* If Zend Thread Safety (ZTS) is defined, each thread gets its own private
- * php_libsmbclient_globals structure, the elements of which it can access
- * through the LIBSMBCLIENT() macro. Without ZTS, there is just one master
+ * php_smbclient_globals structure, the elements of which it can access
+ * through the SMBCLIENT() macro. Without ZTS, there is just one master
  * structure in which we access the members directly: */
 #ifdef ZTS
-#define LIBSMBCLIENT(v) TSRMG(libsmbclient_globals_id, php_libsmbclient_globals *, v)
+#define SMBCLIENT_G(v) TSRMG(smbclient_globals_id, php_smbclient_globals *, v)
 #else
-#define LIBSMBCLIENT(v) (libsmbclient_globals.v)
+#define SMBCLIENT_G(v) (smbclient_globals.v)
 #endif
 
 php_stream_wrapper php_stream_smb_wrapper;
-php_libsmbclient_state * php_libsmbclient_state_new  (php_stream_context *context, int init TSRMLS_DC);
-void                     php_libsmbclient_state_free (php_libsmbclient_state *state TSRMLS_DC);
-int                      php_libsmbclient_state_init (php_libsmbclient_state *state TSRMLS_DC);
-int                      flagstring_to_smbflags (const char *flags, int flags_len, int *retval TSRMLS_DC);
+php_smbclient_state * php_smbclient_state_new  (php_stream_context *context, int init TSRMLS_DC);
+void                  php_smbclient_state_free (php_smbclient_state *state TSRMLS_DC);
+int                   php_smbclient_state_init (php_smbclient_state *state TSRMLS_DC);
+int                   flagstring_to_smbflags (const char *flags, int flags_len, int *retval TSRMLS_DC);
 
 #endif /* PHP_SMBCLIENT_H */

--- a/php_smbclient.h
+++ b/php_smbclient.h
@@ -38,18 +38,18 @@
  * ------------------------------------------------------------------
  */
 
-#ifndef PHP_LIBSMBCLIENT_H
-#define PHP_LIBSMBCLIENT_H
+#ifndef PHP_SMBCLIENT_H
+#define PHP_SMBCLIENT_H
 
 #include <libsmbclient.h>
 
-#define LIBSMBCLIENT_VERSION "0.8.0-dev"
+#define PHP_SMBCLIENT_VERSION "0.8.0-dev"
 
-extern zend_module_entry libsmbclient_module_entry;
+extern zend_module_entry smbclient_module_entry;
 #define phpext_libsmbclient_ptr &libsmbclient_module_entry
 
 typedef struct {
-} php_libsmbclient_globals;
+} php_smbclient_globals;
 
 typedef struct _php_libsmbclient_state
 {
@@ -117,4 +117,4 @@ void                     php_libsmbclient_state_free (php_libsmbclient_state *st
 int                      php_libsmbclient_state_init (php_libsmbclient_state *state TSRMLS_DC);
 int                      flagstring_to_smbflags (const char *flags, int flags_len, int *retval TSRMLS_DC);
 
-#endif /* PHP_LIBSMBCLIENT_H */
+#endif /* PHP_SMBCLIENT_H */

--- a/smb_streams.c
+++ b/smb_streams.c
@@ -46,7 +46,7 @@
 #include "ext/standard/url.h"
 #include "ext/standard/info.h"
 #include "ext/standard/php_filestat.h"
-#include "php_libsmbclient.h"
+#include "php_smbclient.h"
 
 #include <libsmbclient.h>
 

--- a/smb_streams.c
+++ b/smb_streams.c
@@ -54,7 +54,7 @@
 	php_smb_stream_data *self = (php_smb_stream_data *) stream->abstract;
 
 typedef struct _php_smb_stream_data {
-	php_libsmbclient_state *state;
+	php_smbclient_state    *state;
 	SMBCFILE               *handle;
 	/* pointers cache for multiple call */
 	smbc_read_fn            smbc_read;
@@ -81,7 +81,7 @@ static int php_smb_ops_close(php_stream *stream, int close_handle TSRMLS_DC)
 		}
 	}
 
-	php_libsmbclient_state_free(self->state TSRMLS_CC);
+	php_smbclient_state_free(self->state TSRMLS_CC);
 	efree(self);
 	stream->abstract = NULL;
 	return EOF;
@@ -195,7 +195,7 @@ php_stream_smb_opener(
 	php_stream_context *context
 	STREAMS_DC TSRMLS_DC)
 {
-	php_libsmbclient_state *state;
+	php_smbclient_state    *state;
 	int                     smbflags;
 	long                    smbmode = 0666;
 	smbc_open_fn            smbc_open;
@@ -203,7 +203,7 @@ php_stream_smb_opener(
 	php_smb_stream_data    *self;
 
 	/* Context */
-	state = php_libsmbclient_state_new(context, 1 TSRMLS_CC);
+	state = php_smbclient_state_new(context, 1 TSRMLS_CC);
 	if (!state) {
 		return NULL;
 	}
@@ -214,15 +214,15 @@ php_stream_smb_opener(
 		mode = "r";
 	}
 	if (flagstring_to_smbflags(mode, strlen(mode), &smbflags TSRMLS_CC) == 0) {
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return NULL;
 	}
 	if ((smbc_open = smbc_getFunctionOpen(state->ctx)) == NULL) {
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return NULL;
 	}
 	if ((handle = smbc_open(state->ctx, path, smbflags, smbmode)) == NULL) {
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return NULL;
 	}
 	self = ecalloc(sizeof(*self), 1);
@@ -244,11 +244,11 @@ php_stream_smb_unlink(
 	php_stream_context *context
 	TSRMLS_DC)
 {
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 	smbc_unlink_fn smbc_unlink;
 
 	/* Context */
-	state = php_libsmbclient_state_new(context, 1 TSRMLS_CC);
+	state = php_smbclient_state_new(context, 1 TSRMLS_CC);
 	if (!state) {
 		return 0;
 	}
@@ -257,17 +257,17 @@ php_stream_smb_unlink(
 		if (options & REPORT_ERRORS) {
 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Unlink not supported");
 		}
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return 0;
 	}
 	if (smbc_unlink(state->ctx, url) == 0) {
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return 1;
 	}
 	if (options & REPORT_ERRORS) {
 		php_error_docref(NULL TSRMLS_CC, E_WARNING, "Unlink fails: %s", strerror(errno));
 	}
-	php_libsmbclient_state_free(state TSRMLS_CC);
+	php_smbclient_state_free(state TSRMLS_CC);
 	return 0;
 }
 
@@ -284,7 +284,7 @@ php_stream_smb_mkdir(
 	php_stream_context *context
 	TSRMLS_DC)
 {
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 	smbc_mkdir_fn smbc_mkdir;
 
 	if (options & PHP_STREAM_MKDIR_RECURSIVE) {
@@ -292,22 +292,22 @@ php_stream_smb_mkdir(
 		return 0;
 	}
 	/* Context */
-	state = php_libsmbclient_state_new(context, 1 TSRMLS_CC);
+	state = php_smbclient_state_new(context, 1 TSRMLS_CC);
 	if (!state) {
 		return 0;
 	}
 	/* Mkdir */
 	if ((smbc_mkdir = smbc_getFunctionMkdir(state->ctx)) == NULL) {
 		php_error_docref(NULL TSRMLS_CC, E_WARNING, "Mkdir not supported");
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return 0;
 	}
 	if (smbc_mkdir(state->ctx, url, (mode_t)mode) == 0) {
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return 1;
 	}
 	php_error_docref(NULL TSRMLS_CC, E_WARNING, "Mkdir fails: %s", strerror(errno));
-	php_libsmbclient_state_free(state TSRMLS_CC);
+	php_smbclient_state_free(state TSRMLS_CC);
 	return 0;
 }
 
@@ -323,26 +323,26 @@ php_stream_smb_rmdir(
 	php_stream_context *context
 	TSRMLS_DC)
 {
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 	smbc_rmdir_fn smbc_rmdir;
 
 	/* Context */
-	state = php_libsmbclient_state_new(context, 1 TSRMLS_CC);
+	state = php_smbclient_state_new(context, 1 TSRMLS_CC);
 	if (!state) {
 		return 0;
 	}
 	/* Rmdir */
 	if ((smbc_rmdir = smbc_getFunctionRmdir(state->ctx)) == NULL) {
 		php_error_docref(NULL TSRMLS_CC, E_WARNING, "Rmdir not supported");
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return 0;
 	}
 	if (smbc_rmdir(state->ctx, url) == 0) {
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return 1;
 	}
 	php_error_docref(NULL TSRMLS_CC, E_WARNING, "Rmdir fails: %s", strerror(errno));
-	php_libsmbclient_state_free(state TSRMLS_CC);
+	php_smbclient_state_free(state TSRMLS_CC);
 	return 0;
 }
 
@@ -360,25 +360,25 @@ php_stream_smb_rename(
 	php_stream_context *context
 	TSRMLS_DC)
 {
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 	smbc_rename_fn smbc_rename;
 
 	/* Context */
-	state = php_libsmbclient_state_new(context, 1 TSRMLS_CC);
+	state = php_smbclient_state_new(context, 1 TSRMLS_CC);
 	if (!state) {
 		return 0;
 	}
 	if ((smbc_rename = smbc_getFunctionRename(state->ctx)) == NULL) {
 		php_error_docref(NULL TSRMLS_CC, E_WARNING, "Rename not supported");
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return 0;
 	}
 	if (smbc_rename(state->ctx, url_from, state->ctx, url_to) == 0) {
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return 1;
 	}
 	php_error_docref(NULL TSRMLS_CC, E_WARNING, "Rename fails: %s", strerror(errno));
-	php_libsmbclient_state_free(state TSRMLS_CC);
+	php_smbclient_state_free(state TSRMLS_CC);
 	return 0;
 }
 
@@ -396,7 +396,7 @@ static int php_smbdir_ops_close(php_stream *stream, int close_handle TSRMLS_DC)
 			self->handle = NULL;
 		}
 	}
-	php_libsmbclient_state_free(self->state TSRMLS_CC);
+	php_smbclient_state_free(self->state TSRMLS_CC);
 	efree(self);
 	stream->abstract = NULL;
 	return EOF;
@@ -459,23 +459,23 @@ php_stream_smbdir_opener(
 	php_stream_context *context
 	STREAMS_DC TSRMLS_DC)
 {
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 	smbc_opendir_fn         smbc_opendir;
 	SMBCFILE               *handle;
 	php_smb_stream_data    *self;
 
 	/* Context */
-	state = php_libsmbclient_state_new(context, 1 TSRMLS_CC);
+	state = php_smbclient_state_new(context, 1 TSRMLS_CC);
 	if (!state) {
 		return NULL;
 	}
 	/* Directory */
 	if ((smbc_opendir = smbc_getFunctionOpendir(state->ctx)) == NULL) {
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return NULL;
 	}
 	if ((handle = smbc_opendir(state->ctx, path)) == NULL) {
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return NULL;
 	}
 	self = ecalloc(sizeof(*self), 1);
@@ -498,26 +498,26 @@ php_stream_smb_stat(
 	php_stream_context *context
 	TSRMLS_DC)
 {
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 	smbc_stat_fn smbc_stat;
 
 	/* Context */
-	state = php_libsmbclient_state_new(context, 1 TSRMLS_CC);
+	state = php_smbclient_state_new(context, 1 TSRMLS_CC);
 	if (!state) {
 		return 0;
 	}
 	/* Stat */
 	if ((smbc_stat = smbc_getFunctionStat(state->ctx)) == NULL) {
 		php_error_docref(NULL TSRMLS_CC, E_WARNING, "Stat not supported");
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return -1;
 	}
 	if (smbc_stat(state->ctx, url, &ssb->sb) >= 0) {
-		php_libsmbclient_state_free(state TSRMLS_CC);
+		php_smbclient_state_free(state TSRMLS_CC);
 		return 0;
 	}
 	/* dont display error as PHP use this method internally to check if file exists */
-	php_libsmbclient_state_free(state TSRMLS_CC);
+	php_smbclient_state_free(state TSRMLS_CC);
 	return -1;
 }
 
@@ -535,7 +535,7 @@ php_stream_smb_metadata(
 	php_stream_context *context
 	TSRMLS_DC)
 {
-	php_libsmbclient_state *state;
+	php_smbclient_state    *state;
 	smbc_chmod_fn           smbc_chmod;
 	smbc_open_fn            smbc_open;
 	smbc_utimes_fn          smbc_utimes;
@@ -551,7 +551,7 @@ php_stream_smb_metadata(
 			newtime = (struct utimbuf *)value;
 
 			/* Context */
-			state = php_libsmbclient_state_new(context, 1 TSRMLS_CC);
+			state = php_smbclient_state_new(context, 1 TSRMLS_CC);
 			if (!state) {
 				return 0;
 			}
@@ -580,7 +580,7 @@ php_stream_smb_metadata(
 		case PHP_STREAM_META_ACCESS:
 			mode = (mode_t)*(long *)value;
 			/* Context */
-			state = php_libsmbclient_state_new(context, 1 TSRMLS_CC);
+			state = php_smbclient_state_new(context, 1 TSRMLS_CC);
 			if (!state) {
 				return 0;
 			}
@@ -596,7 +596,7 @@ php_stream_smb_metadata(
 			php_error_docref1(NULL TSRMLS_CC, url, E_WARNING, "Unknown option %d for stream_metadata", option);
 			return 0;
 	}
-	php_libsmbclient_state_free(state TSRMLS_CC);
+	php_smbclient_state_free(state TSRMLS_CC);
 	if (ret == -1) {
 		php_error_docref1(NULL TSRMLS_CC, url, E_WARNING, "Operation failed: %s", strerror(errno));
 		return 0;

--- a/smbclient.c
+++ b/smbclient.c
@@ -49,7 +49,7 @@
 #include "php_smbclient.h"
 
 /* If Zend Thread Safety (ZTS) is defined, each thread gets its own copy of
- * the php_libsmbclient_globals structure. Else we use a single global copy: */
+ * the php_smbclient_globals structure. Else we use a single global copy: */
 #ifdef ZTS
 static int smbclient_globals_id;
 #else
@@ -62,11 +62,11 @@ static void php_smbclient_init_globals(php_smbclient_globals *smbclient_globals_
 	 * We currently don't use this. */
 }
 
-#define PHP_LIBSMBCLIENT_STATE_NAME "smbclient state"
-#define PHP_LIBSMBCLIENT_FILE_NAME "smbclient file"
+#define PHP_SMBCLIENT_STATE_NAME "smbclient state"
+#define PHP_SMBCLIENT_FILE_NAME "smbclient file"
 
-static int le_libsmbclient_state;
-static int le_libsmbclient_file;
+static int le_smbclient_state;
+static int le_smbclient_file;
 
 #if PHP_MAJOR_VERSION >= 7
 typedef size_t strsize_t;
@@ -95,7 +95,7 @@ enum {
 	SMBCLIENT_OPT_PORT = 14,
 	SMBCLIENT_OPT_TIMEOUT = 15,
 }
-php_libsmbclient_options;
+php_smbclient_options;
 
 static char *
 find_char (char *start, char *last, char q)
@@ -360,7 +360,7 @@ smbclient_auth_func (SMBCCTX *ctx, const char *server, const char *share, char *
 	/* Given context, server and share, return workgroup, username and password.
 	 * String lengths are the max allowable lengths. */
 
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (ctx == NULL || (state = smbc_getOptionUserData(ctx)) == NULL) {
 		return;
@@ -371,7 +371,7 @@ smbclient_auth_func (SMBCCTX *ctx, const char *server, const char *share, char *
 }
 
 void
-php_libsmbclient_state_free (php_libsmbclient_state *state TSRMLS_DC)
+php_smbclient_state_free (php_smbclient_state *state TSRMLS_DC)
 {
 	/* TODO: if smbc_free_context() returns 0, PHP will leak the handle: */
 	if (state->ctx != NULL && smbc_free_context(state->ctx, 1) != 0) {
@@ -405,7 +405,7 @@ smbclient_state_dtor (
 #endif
 	TSRMLS_DC)
 {
-	php_libsmbclient_state_free((php_libsmbclient_state *)rsrc->ptr TSRMLS_CC);
+	php_smbclient_state_free((php_smbclient_state *)rsrc->ptr TSRMLS_CC);
 }
 
 static void
@@ -479,8 +479,8 @@ PHP_MINIT_FUNCTION(smbclient)
 	REGISTER_LONG_CONSTANT("SMBCLIENT_VFS_CASE_INSENSITIVE", SMBC_VFS_FEATURE_CASE_INSENSITIVE, CONST_PERSISTENT | CONST_CS);
 	REGISTER_LONG_CONSTANT("SMBCLIENT_VFS_NO_UNIXCIFS", SMBC_VFS_FEATURE_NO_UNIXCIFS, CONST_PERSISTENT | CONST_CS);
 
-	le_libsmbclient_state = zend_register_list_destructors_ex(smbclient_state_dtor, NULL, PHP_LIBSMBCLIENT_STATE_NAME, module_number);
-	le_libsmbclient_file = zend_register_list_destructors_ex(smbclient_file_dtor, NULL, PHP_LIBSMBCLIENT_FILE_NAME, module_number);
+	le_smbclient_state = zend_register_list_destructors_ex(smbclient_state_dtor, NULL, PHP_SMBCLIENT_STATE_NAME, module_number);
+	le_smbclient_file = zend_register_list_destructors_ex(smbclient_file_dtor, NULL, PHP_SMBCLIENT_FILE_NAME, module_number);
 
 	php_register_url_stream_wrapper("smb", &php_stream_smb_wrapper TSRMLS_CC);
 
@@ -553,10 +553,10 @@ ctx_init_getauth (zval *z, char **dest, int *destlen, char *varname)
 	}
 }
 
-php_libsmbclient_state *
-php_libsmbclient_state_new (php_stream_context *context, int init TSRMLS_DC)
+php_smbclient_state *
+php_smbclient_state_new (php_stream_context *context, int init TSRMLS_DC)
 {
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 	SMBCCTX *ctx;
 
 	if ((ctx = smbc_new_context()) == NULL) {
@@ -566,7 +566,7 @@ php_libsmbclient_state_new (php_stream_context *context, int init TSRMLS_DC)
 		};
 		return NULL;
 	}
-	state = emalloc(sizeof(php_libsmbclient_state));
+	state = emalloc(sizeof(php_smbclient_state));
 	state->ctx = ctx;
 	state->wrkg = NULL;
 	state->user = NULL;
@@ -591,19 +591,19 @@ php_libsmbclient_state_new (php_stream_context *context, int init TSRMLS_DC)
 
 		if (NULL != (tmpzval = php_stream_context_get_option(context, "smb", "workgroup"))) {
 			if (ctx_init_getauth(tmpzval, &state->wrkg, &state->wrkglen, "workgroup") == 0) {
-				php_libsmbclient_state_free(state);
+				php_smbclient_state_free(state);
 				return NULL;
 			}
 		}
 		if (NULL != (tmpzval = php_stream_context_get_option(context, "smb", "username"))) {
 			if (ctx_init_getauth(tmpzval, &state->user, &state->userlen, "username") == 0) {
-				php_libsmbclient_state_free(state);
+				php_smbclient_state_free(state);
 				return NULL;
 			}
 		}
 		if (NULL != (tmpzval = php_stream_context_get_option(context, "smb", "password"))) {
 			if (ctx_init_getauth(tmpzval, &state->pass, &state->passlen, "password") == 0) {
-				php_libsmbclient_state_free(state);
+				php_smbclient_state_free(state);
 				return NULL;
 			}
 		}
@@ -612,27 +612,27 @@ php_libsmbclient_state_new (php_stream_context *context, int init TSRMLS_DC)
 
 		if (php_stream_context_get_option(context, "smb", "workgroup", &tmpzval) == SUCCESS) {
 			if (ctx_init_getauth(*tmpzval, &state->wrkg, &state->wrkglen, "workgroup") == 0) {
-				php_libsmbclient_state_free(state TSRMLS_CC);
+				php_smbclient_state_free(state TSRMLS_CC);
 				return NULL;
 			}
 		}
 		if (php_stream_context_get_option(context, "smb", "username", &tmpzval) == SUCCESS) {
 			if (ctx_init_getauth(*tmpzval, &state->user, &state->userlen, "username") == 0) {
-				php_libsmbclient_state_free(state TSRMLS_CC);
+				php_smbclient_state_free(state TSRMLS_CC);
 				return NULL;
 			}
 		}
 		if (php_stream_context_get_option(context, "smb", "password", &tmpzval) == SUCCESS) {
 			if (ctx_init_getauth(*tmpzval, &state->pass, &state->passlen, "password") == 0) {
-				php_libsmbclient_state_free(state TSRMLS_CC);
+				php_smbclient_state_free(state TSRMLS_CC);
 				return NULL;
 			}
 		}
 #endif
 	}
 	if (init) {
-		if (php_libsmbclient_state_init(state TSRMLS_CC)) {
-			php_libsmbclient_state_free(state TSRMLS_CC);
+		if (php_smbclient_state_init(state TSRMLS_CC)) {
+			php_smbclient_state_free(state TSRMLS_CC);
 			return NULL;
 		}
 	}
@@ -641,18 +641,18 @@ php_libsmbclient_state_new (php_stream_context *context, int init TSRMLS_DC)
 
 PHP_FUNCTION(smbclient_state_new)
 {
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters_none() == FAILURE) {
 		RETURN_FALSE;
 	}
-	if ((state = php_libsmbclient_state_new(NULL, 0 TSRMLS_CC)) == NULL) {
+	if ((state = php_smbclient_state_new(NULL, 0 TSRMLS_CC)) == NULL) {
 		RETURN_FALSE;
 	}
 #if PHP_MAJOR_VERSION >= 7
-	ZVAL_RES(return_value, zend_register_resource(state, le_libsmbclient_state));
+	ZVAL_RES(return_value, zend_register_resource(state, le_smbclient_state));
 #else
-	ZEND_REGISTER_RESOURCE(return_value, state, le_libsmbclient_state);
+	ZEND_REGISTER_RESOURCE(return_value, state, le_smbclient_state);
 #endif
 }
 
@@ -681,7 +681,7 @@ PHP_FUNCTION(smbclient_library_version)
 }
 
 int
-php_libsmbclient_state_init (php_libsmbclient_state *state TSRMLS_DC)
+php_smbclient_state_init (php_smbclient_state *state TSRMLS_DC)
 {
 	SMBCCTX *ctx;
 
@@ -712,16 +712,16 @@ php_libsmbclient_state_init (php_libsmbclient_state *state TSRMLS_DC)
 #endif
 
 #define STATE_FROM_ZSTATE \
-	SMB_FETCH_RESOURCE(state, php_libsmbclient_state *, &zstate, PHP_LIBSMBCLIENT_STATE_NAME, le_libsmbclient_state); \
+	SMB_FETCH_RESOURCE(state, php_smbclient_state *, &zstate, PHP_SMBCLIENT_STATE_NAME, le_smbclient_state); \
 	if (state == NULL || state->ctx == NULL) { \
-		php_error(E_WARNING, PHP_LIBSMBCLIENT_STATE_NAME " not found"); \
+		php_error(E_WARNING, PHP_SMBCLIENT_STATE_NAME " not found"); \
 		RETURN_FALSE; \
 	}
 
 #define FILE_FROM_ZFILE \
-	SMB_FETCH_RESOURCE(file, SMBCFILE *, &zfile, PHP_LIBSMBCLIENT_FILE_NAME, le_libsmbclient_file); \
+	SMB_FETCH_RESOURCE(file, SMBCFILE *, &zfile, PHP_SMBCLIENT_FILE_NAME, le_smbclient_file); \
 	if (file == NULL) { \
-		php_error(E_WARNING, PHP_LIBSMBCLIENT_FILE_NAME " not found"); \
+		php_error(E_WARNING, PHP_SMBCLIENT_FILE_NAME " not found"); \
 		RETURN_FALSE; \
 	}
 
@@ -731,12 +731,12 @@ PHP_FUNCTION(smbclient_state_init)
 	zval *zwrkg = NULL;
 	zval *zuser = NULL;
 	zval *zpass = NULL;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "r|zzz", &zstate, &zwrkg, &zuser, &zpass) != SUCCESS) {
 		RETURN_FALSE;
 	}
-	SMB_FETCH_RESOURCE(state, php_libsmbclient_state *, &zstate, PHP_LIBSMBCLIENT_STATE_NAME, le_libsmbclient_state);
+	SMB_FETCH_RESOURCE(state, php_smbclient_state *, &zstate, PHP_SMBCLIENT_STATE_NAME, le_smbclient_state);
 
 	if (state->ctx == NULL) {
 		php_error(E_WARNING, "Couldn't init SMB context: context is null");
@@ -751,7 +751,7 @@ PHP_FUNCTION(smbclient_state_init)
 	if (ctx_init_getauth(zpass, &state->pass, &state->passlen, "password") == 0) {
 		RETURN_FALSE;
 	}
-	if (php_libsmbclient_state_init(state TSRMLS_CC)) {
+	if (php_smbclient_state_init(state TSRMLS_CC)) {
 		RETURN_FALSE;
 	}
 	RETURN_TRUE;
@@ -760,24 +760,24 @@ PHP_FUNCTION(smbclient_state_init)
 PHP_FUNCTION(smbclient_state_errno)
 {
 	zval *zstate;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "r", &zstate) != SUCCESS) {
 		RETURN_LONG(0);
 	}
-	SMB_FETCH_RESOURCE(state, php_libsmbclient_state *, &zstate, PHP_LIBSMBCLIENT_STATE_NAME, le_libsmbclient_state);
+	SMB_FETCH_RESOURCE(state, php_smbclient_state *, &zstate, PHP_SMBCLIENT_STATE_NAME, le_smbclient_state);
 	RETURN_LONG(state->err);
 }
 
 PHP_FUNCTION(smbclient_state_free)
 {
 	zval *zstate;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "r", &zstate) != SUCCESS) {
 		RETURN_FALSE;
 	}
-	SMB_FETCH_RESOURCE(state, php_libsmbclient_state *, &zstate, PHP_LIBSMBCLIENT_STATE_NAME, le_libsmbclient_state);
+	SMB_FETCH_RESOURCE(state, php_smbclient_state *, &zstate, PHP_SMBCLIENT_STATE_NAME, le_smbclient_state);
 
 	if (state->ctx == NULL) {
 #if PHP_MAJOR_VERSION >= 7
@@ -811,7 +811,7 @@ PHP_FUNCTION(smbclient_opendir)
 	zval *zstate;
 	SMBCFILE *dir;
 	smbc_opendir_fn smbc_opendir;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rs", &zstate, &path, &path_len) == FAILURE) {
 		return;
@@ -823,9 +823,9 @@ PHP_FUNCTION(smbclient_opendir)
 	}
 	if ((dir = smbc_opendir(state->ctx, path)) != NULL) {
 #if PHP_MAJOR_VERSION >= 7
-		ZVAL_RES(return_value, zend_register_resource(dir, le_libsmbclient_file));
+		ZVAL_RES(return_value, zend_register_resource(dir, le_smbclient_file));
 #else
-		ZEND_REGISTER_RESOURCE(return_value, dir, le_libsmbclient_file);
+		ZEND_REGISTER_RESOURCE(return_value, dir, le_smbclient_file);
 #endif
 		return;
 	}
@@ -867,7 +867,7 @@ PHP_FUNCTION(smbclient_readdir)
 	zval *zfile;
 	SMBCFILE *file;
 	smbc_readdir_fn smbc_readdir;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rr", &zstate, &zfile) == FAILURE) {
 		return;
@@ -882,9 +882,9 @@ PHP_FUNCTION(smbclient_readdir)
 	if ((dirent = smbc_readdir(state->ctx, file)) == NULL) {
 		switch (state->err = errno) {
 			case 0: RETURN_FALSE;
-			case EBADF: php_error(E_WARNING, "Couldn't read " PHP_LIBSMBCLIENT_FILE_NAME ": Not a directory resource"); break;
-			case EINVAL: php_error(E_WARNING, "Couldn't read " PHP_LIBSMBCLIENT_FILE_NAME ": State resource not initialized"); break;
-			default: php_error(E_WARNING, "Couldn't read " PHP_LIBSMBCLIENT_FILE_NAME ": unknown error (%d)", errno); break;
+			case EBADF: php_error(E_WARNING, "Couldn't read " PHP_SMBCLIENT_FILE_NAME ": Not a directory resource"); break;
+			case EINVAL: php_error(E_WARNING, "Couldn't read " PHP_SMBCLIENT_FILE_NAME ": State resource not initialized"); break;
+			default: php_error(E_WARNING, "Couldn't read " PHP_SMBCLIENT_FILE_NAME ": unknown error (%d)", errno); break;
 		}
 		RETURN_FALSE;
 	}
@@ -909,7 +909,7 @@ PHP_FUNCTION(smbclient_closedir)
 	zval *zfile;
 	SMBCFILE *file;
 	smbc_closedir_fn smbc_closedir;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rr", &zstate, &zfile) == FAILURE) {
 		return;
@@ -929,8 +929,8 @@ PHP_FUNCTION(smbclient_closedir)
 		RETURN_TRUE;
 	}
 	switch (state->err = errno) {
-		case EBADF: php_error(E_WARNING, "Couldn't close " PHP_LIBSMBCLIENT_FILE_NAME ": Not a directory resource"); break;
-		default: php_error(E_WARNING, "Couldn't close " PHP_LIBSMBCLIENT_FILE_NAME ": unknown error (%d)", errno); break;
+		case EBADF: php_error(E_WARNING, "Couldn't close " PHP_SMBCLIENT_FILE_NAME ": Not a directory resource"); break;
+		default: php_error(E_WARNING, "Couldn't close " PHP_SMBCLIENT_FILE_NAME ": unknown error (%d)", errno); break;
 	}
 	RETURN_FALSE;
 }
@@ -942,21 +942,21 @@ PHP_FUNCTION(smbclient_rename)
 	zval *zstate_old;
 	zval *zstate_new;
 	smbc_rename_fn smbc_rename;
-	php_libsmbclient_state *state_old;
-	php_libsmbclient_state *state_new;
+	php_smbclient_state *state_old;
+	php_smbclient_state *state_new;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rsrs", &zstate_old, &ourl, &ourl_len, &zstate_new, &nurl, &nurl_len) == FAILURE) {
 		return;
 	}
-	SMB_FETCH_RESOURCE(state_old, php_libsmbclient_state *, &zstate_old, PHP_LIBSMBCLIENT_STATE_NAME, le_libsmbclient_state);
-	SMB_FETCH_RESOURCE(state_new, php_libsmbclient_state *, &zstate_new, PHP_LIBSMBCLIENT_STATE_NAME, le_libsmbclient_state);
+	SMB_FETCH_RESOURCE(state_old, php_smbclient_state *, &zstate_old, PHP_SMBCLIENT_STATE_NAME, le_smbclient_state);
+	SMB_FETCH_RESOURCE(state_new, php_smbclient_state *, &zstate_new, PHP_SMBCLIENT_STATE_NAME, le_smbclient_state);
 
 	if (state_old == NULL || state_old->ctx == NULL) {
-		php_error(E_WARNING, "old " PHP_LIBSMBCLIENT_STATE_NAME " is null");
+		php_error(E_WARNING, "old " PHP_SMBCLIENT_STATE_NAME " is null");
 		RETURN_FALSE;
 	}
 	if (state_new == NULL || state_new->ctx == NULL) {
-		php_error(E_WARNING, "new " PHP_LIBSMBCLIENT_STATE_NAME " is null");
+		php_error(E_WARNING, "new " PHP_SMBCLIENT_STATE_NAME " is null");
 		RETURN_FALSE;
 	}
 	if ((smbc_rename = smbc_getFunctionRename(state_old->ctx)) == NULL) {
@@ -987,7 +987,7 @@ PHP_FUNCTION(smbclient_unlink)
 	strsize_t url_len;
 	zval *zstate;
 	smbc_unlink_fn smbc_unlink;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rs", &zstate, &url, &url_len) == FAILURE) {
 		return;
@@ -1021,7 +1021,7 @@ PHP_FUNCTION(smbclient_mkdir)
 	zend_long mode = 0777;	/* Same as PHP's native mkdir() */
 	zval *zstate;
 	smbc_mkdir_fn smbc_mkdir;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rs|l", &zstate, &path, &path_len, &mode) == FAILURE) {
 		return;
@@ -1052,7 +1052,7 @@ PHP_FUNCTION(smbclient_rmdir)
 	strsize_t url_len;
 	zval *zstate;
 	smbc_rmdir_fn smbc_rmdir;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rs", &zstate, &url, &url_len) == FAILURE) {
 		return;
@@ -1085,7 +1085,7 @@ PHP_FUNCTION(smbclient_stat)
 	strsize_t file_len;
 	zval *zstate;
 	smbc_stat_fn smbc_stat;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rs", &zstate, &file, &file_len) == FAILURE) {
 		return;
@@ -1147,7 +1147,7 @@ PHP_FUNCTION(smbclient_fstat)
 	zval *zfile;
 	SMBCFILE *file;
 	smbc_fstat_fn smbc_fstat;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rr", &zstate, &zfile) == FAILURE) {
 		return;
@@ -1160,12 +1160,12 @@ PHP_FUNCTION(smbclient_fstat)
 	}
 	if (smbc_fstat(state->ctx, file, &statbuf) < 0) {
 		switch (state->err = errno) {
-			case ENOENT: php_error(E_WARNING, "Couldn't fstat " PHP_LIBSMBCLIENT_FILE_NAME ": Does not exist"); break;
+			case ENOENT: php_error(E_WARNING, "Couldn't fstat " PHP_SMBCLIENT_FILE_NAME ": Does not exist"); break;
 			case EINVAL: php_error(E_WARNING, "Couldn't fstat: null resource or smbc_init failed"); break;
-			case EACCES: php_error(E_WARNING, "Couldn't fstat " PHP_LIBSMBCLIENT_FILE_NAME ": Permission denied"); break;
-			case ENOMEM: php_error(E_WARNING, "Couldn't fstat " PHP_LIBSMBCLIENT_FILE_NAME ": Out of memory"); break;
-			case ENOTDIR: php_error(E_WARNING, "Couldn't fstat " PHP_LIBSMBCLIENT_FILE_NAME ": Not a directory"); break;
-			default: php_error(E_WARNING, "Couldn't fstat " PHP_LIBSMBCLIENT_FILE_NAME ": unknown error (%d)", errno); break;
+			case EACCES: php_error(E_WARNING, "Couldn't fstat " PHP_SMBCLIENT_FILE_NAME ": Permission denied"); break;
+			case ENOMEM: php_error(E_WARNING, "Couldn't fstat " PHP_SMBCLIENT_FILE_NAME ": Out of memory"); break;
+			case ENOTDIR: php_error(E_WARNING, "Couldn't fstat " PHP_SMBCLIENT_FILE_NAME ": Not a directory"); break;
+			default: php_error(E_WARNING, "Couldn't fstat " PHP_SMBCLIENT_FILE_NAME ": unknown error (%d)", errno); break;
 		}
 		RETURN_FALSE;
 	}
@@ -1246,7 +1246,7 @@ PHP_FUNCTION(smbclient_open)
 	zval *zstate;
 	SMBCFILE *handle;
 	smbc_open_fn smbc_open;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rss|l", &zstate, &file, &file_len, &flags, &flags_len, &mode) == FAILURE) {
 		return;
@@ -1264,9 +1264,9 @@ PHP_FUNCTION(smbclient_open)
 	}
 	if ((handle = smbc_open(state->ctx, file, smbflags, mode)) != NULL) {
 #if PHP_MAJOR_VERSION >= 7
-		ZVAL_RES(return_value, zend_register_resource(handle, le_libsmbclient_file));
+		ZVAL_RES(return_value, zend_register_resource(handle, le_smbclient_file));
 #else
-		ZEND_REGISTER_RESOURCE(return_value, handle, le_libsmbclient_file);
+		ZEND_REGISTER_RESOURCE(return_value, handle, le_smbclient_file);
 #endif
 		return;
 	}
@@ -1293,7 +1293,7 @@ PHP_FUNCTION(smbclient_creat)
 	zval *zstate;
 	SMBCFILE *handle;
 	smbc_creat_fn smbc_creat;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rs|l", &zstate, &file, &file_len, &mode) == FAILURE) {
 		return;
@@ -1305,9 +1305,9 @@ PHP_FUNCTION(smbclient_creat)
 	}
 	if ((handle = smbc_creat(state->ctx, file, (mode_t)mode)) != NULL) {
 #if PHP_MAJOR_VERSION >= 7
-		ZVAL_RES(return_value, zend_register_resource(handle, le_libsmbclient_file));
+		ZVAL_RES(return_value, zend_register_resource(handle, le_smbclient_file));
 #else
-		ZEND_REGISTER_RESOURCE(return_value, handle, le_libsmbclient_file);
+		ZEND_REGISTER_RESOURCE(return_value, handle, le_smbclient_file);
 #endif
 
 		return;
@@ -1333,7 +1333,7 @@ PHP_FUNCTION(smbclient_read)
 	zval *zfile;
 	SMBCFILE *file;
 	smbc_read_fn smbc_read;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rrl", &zstate, &zfile, &count) == FAILURE) {
 		return;
@@ -1382,7 +1382,7 @@ PHP_FUNCTION(smbclient_write)
 	zval *zfile;
 	SMBCFILE *file;
 	smbc_write_fn smbc_write;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rrs|l", &zstate, &zfile, &str, &str_len, &count) == FAILURE) {
 		return;
@@ -1423,7 +1423,7 @@ PHP_FUNCTION(smbclient_lseek)
 	zval *zfile;
 	SMBCFILE *file;
 	smbc_lseek_fn smbc_lseek;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rrll", &zstate, &zfile, &offset, &whence) == FAILURE) {
 		return;
@@ -1456,7 +1456,7 @@ PHP_FUNCTION(smbclient_ftruncate)
 	zval *zfile;
 	SMBCFILE *file;
 	smbc_ftruncate_fn smbc_ftruncate;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rrl", &zstate, &zfile, &offset) == FAILURE) {
 		return;
@@ -1486,7 +1486,7 @@ PHP_FUNCTION(smbclient_close)
 	zval *zfile;
 	SMBCFILE *file;
 	smbc_close_fn smbc_close;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rr", &zstate, &zfile) == FAILURE) {
 		return;
@@ -1520,7 +1520,7 @@ PHP_FUNCTION(smbclient_chmod)
 	zend_long mode;
 	zval *zstate;
 	smbc_chmod_fn smbc_chmod;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rsl", &zstate, &file, &file_len, &mode) == FAILURE) {
 		return;
@@ -1551,7 +1551,7 @@ PHP_FUNCTION(smbclient_utimes)
 	zval *zstate;
 	struct timeval times[2];
 	smbc_utimes_fn smbc_utimes;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rs|ll", &zstate, &file, &file_len, &mtime, &atime) == FAILURE) {
 		return;
@@ -1591,7 +1591,7 @@ PHP_FUNCTION(smbclient_listxattr)
 	char values[1000];
 	zval *zstate;
 	smbc_listxattr_fn smbc_listxattr;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rs", &zstate, &url, &url_len) == FAILURE) {
 		return;
@@ -1648,7 +1648,7 @@ PHP_FUNCTION(smbclient_getxattr)
 	char values[1000];
 	zval *zstate;
 	smbc_getxattr_fn smbc_getxattr;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rss", &zstate, &url, &url_len, &name, &name_len) == FAILURE) {
 		return;
@@ -1690,7 +1690,7 @@ PHP_FUNCTION(smbclient_setxattr)
 	zend_long flags = 0;
 	zval *zstate;
 	smbc_setxattr_fn smbc_setxattr;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rsss|l", &zstate, &url, &url_len, &name, &name_len, &val, &val_len, &flags) == FAILURE) {
 		return;
@@ -1722,7 +1722,7 @@ PHP_FUNCTION(smbclient_removexattr)
 	strsize_t url_len, name_len;
 	zval *zstate;
 	smbc_removexattr_fn smbc_removexattr;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rss", &zstate, &url, &url_len, &name, &name_len) == FAILURE) {
 		return;
@@ -1751,7 +1751,7 @@ PHP_FUNCTION(smbclient_option_get)
 	zend_long option;
 	char *ret;
 	zval *zstate;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rl", &zstate, &option) == FAILURE) {
 		return;
@@ -1848,7 +1848,7 @@ PHP_FUNCTION(smbclient_option_set)
 	zend_long option, vbool = 0;
 	zval *zstate;
 	zval *zvalue;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rlz", &zstate, &option, &zvalue) == FAILURE) {
 		return;
@@ -1964,7 +1964,7 @@ PHP_FUNCTION(smbclient_statvfs)
 	zval *zstate;
 	struct statvfs st;
 	smbc_statvfs_fn smbc_statvfs;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rs", &zstate, &url, &url_len) == FAILURE) {
 		return;
@@ -2009,7 +2009,7 @@ PHP_FUNCTION(smbclient_fstatvfs)
 	SMBCFILE *file;
 	struct statvfs st;
 	smbc_fstatvfs_fn smbc_fstatvfs;
-	php_libsmbclient_state *state;
+	php_smbclient_state *state;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "rr", &zstate, &zfile) == FAILURE) {
 		return;

--- a/smbclient.c
+++ b/smbclient.c
@@ -46,17 +46,17 @@
 
 #include "php.h"
 #include "ext/standard/info.h"
-#include "php_libsmbclient.h"
+#include "php_smbclient.h"
 
 /* If Zend Thread Safety (ZTS) is defined, each thread gets its own copy of
  * the php_libsmbclient_globals structure. Else we use a single global copy: */
 #ifdef ZTS
-static int libsmbclient_globals_id;
+static int smbclient_globals_id;
 #else
-static php_libsmbclient_globals libsmbclient_globals;
+static php_smbclient_globals smbclient_globals;
 #endif
 
-static void php_libsmbclient_init_globals(php_libsmbclient_globals *libsmbclient_globals_p TSRMLS_DC)
+static void php_smbclient_init_globals(php_smbclient_globals *smbclient_globals_p TSRMLS_DC)
 {
 	/* This function initializes the thread-local storage.
 	 * We currently don't use this. */
@@ -263,7 +263,7 @@ ZEND_END_ARG_INFO()
 
 /* }}} */
 
-static zend_function_entry libsmbclient_functions[] =
+static zend_function_entry smbclient_functions[] =
 {
 	PHP_FE(smbclient_version, arginfo_smbclient_void)
 	PHP_FE(smbclient_library_version, arginfo_smbclient_void)
@@ -304,21 +304,35 @@ static zend_function_entry libsmbclient_functions[] =
 #endif
 };
 
-zend_module_entry libsmbclient_module_entry =
+zend_module_entry smbclient_module_entry =
 	{ STANDARD_MODULE_HEADER
-	, "libsmbclient"		/* name                  */
-	, libsmbclient_functions	/* functions             */
+	, "smbclient"		/* name                  */
+	, smbclient_functions	/* functions             */
 	, PHP_MINIT(smbclient)		/* module_startup_func   */
 	, PHP_MSHUTDOWN(smbclient)	/* module_shutdown_func  */
 	, PHP_RINIT(smbclient)		/* request_startup_func  */
 	, NULL				/* request_shutdown_func */
 	, PHP_MINFO(smbclient)		/* info_func             */
-	, LIBSMBCLIENT_VERSION		/* version               */
+	, PHP_SMBCLIENT_VERSION		/* version               */
 	, STANDARD_MODULE_PROPERTIES
 	} ;
 
-#ifdef COMPILE_DL_LIBSMBCLIENT
-ZEND_GET_MODULE(libsmbclient)
+zend_module_entry old_module_entry = {
+	STANDARD_MODULE_HEADER,
+	"libsmbclient",
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	NULL,
+	PHP_SMBCLIENT_VERSION,
+	STANDARD_MODULE_PROPERTIES,
+};
+
+
+#ifdef COMPILE_DL_SMBCLIENT
+ZEND_GET_MODULE(smbclient)
 #endif
 
 static void
@@ -418,9 +432,9 @@ PHP_MINIT_FUNCTION(smbclient)
 	/* If ZTS is defined, allocate and init a copy of the global
 	 * datastructure for each thread: */
 	#ifdef ZTS
-	ts_allocate_id(&libsmbclient_globals_id, sizeof(php_libsmbclient_globals), (ts_allocate_ctor) php_libsmbclient_init_globals, NULL);
+	ts_allocate_id(&smbclient_globals_id, sizeof(php_smbclient_globals), (ts_allocate_ctor) php_smbclient_init_globals, NULL);
 	#else
-	php_libsmbclient_init_globals(&libsmbclient_globals);
+	php_smbclient_init_globals(&smbclient_globals);
 	#endif
 
 	/* Constants for smbclient_setxattr: */
@@ -470,6 +484,9 @@ PHP_MINIT_FUNCTION(smbclient)
 
 	php_register_url_stream_wrapper("smb", &php_stream_smb_wrapper TSRMLS_CC);
 
+	/* register with old name for code using extension_loaded(libsmbclient) */
+	zend_register_internal_module(&old_module_entry TSRMLS_CC);
+
 	return SUCCESS;
 }
 
@@ -486,8 +503,8 @@ PHP_MSHUTDOWN_FUNCTION(smbclient)
 PHP_MINFO_FUNCTION(smbclient)
 {
 	php_info_print_table_start();
-	php_info_print_table_row(2, "libsmbclient Support", "enabled");
-	php_info_print_table_row(2, "libsmbclient extension Version", LIBSMBCLIENT_VERSION);
+	php_info_print_table_row(2, "smbclient Support", "enabled");
+	php_info_print_table_row(2, "smbclient extension Version", PHP_SMBCLIENT_VERSION);
 	php_info_print_table_row(2, "libsmbclient library Version", smbc_version());
 	php_info_print_table_end();
 }
@@ -645,9 +662,9 @@ PHP_FUNCTION(smbclient_version)
 		RETURN_FALSE;
 	}
 #if PHP_MAJOR_VERSION >= 7
-	RETURN_STRING(LIBSMBCLIENT_VERSION);
+	RETURN_STRING(PHP_SMBCLIENT_VERSION);
 #else
-	RETURN_STRING(LIBSMBCLIENT_VERSION, 1);
+	RETURN_STRING(PHP_SMBCLIENT_VERSION, 1);
 #endif
 }
 

--- a/tests/StreamsTest.php
+++ b/tests/StreamsTest.php
@@ -194,4 +194,12 @@ class StreamsTest extends PHPUnit_Framework_TestCase
 		$fi = new finfo(FILEINFO_MIME_TYPE);
 		$this->assertEquals('text/plain', $fi->file($this->readuri));
 	}
+
+	public function
+	testListShares ()
+	{
+		$smb = scandir('smb://'.SMB_USER.':'.SMB_PASS.'@'.SMB_HOST.'/');
+
+		$this->assertContains(SMB_SHARE, $smb, print_r($smb, true));
+	}
 }

